Api/Services/UserService.cs

```diff
@@ -188,65 +188,61 @@ public async Task<bool> IsUniqueLoginAsync(string login)
-            .Select(u => new UserEmployeeVM
+            .Include(u => u.Employments)
+            .ThenInclude(e => e.Restaurant)
+            .Select(u => new
             {
-                UserId = u.Id,
-                Login = u.UserName!,
-                FirstName = u.FirstName,
-                LastName = u.LastName,
-                BirthDate = u.BirthDate!.Value,
-                PhoneNumber = u.PhoneNumber!,
-                Employments = u.Employments
+                u.Id,
+                u.UserName,
+                u.FirstName,
+                u.LastName,
+                u.BirthDate,
+                u.PhoneNumber,
+                u.Employments,
+                u.PhotoFileName
+            })
+            .ToListAsync();
+
+        var employeeVMs = new List<UserEmployeeVM>();
+
+        foreach (var user in users)
+        {
+            var employeeVM = new UserEmployeeVM
+            {
+                UserId = user.Id,
+                Login = user.UserName!,
+                FirstName = user.FirstName,
+                LastName = user.LastName,
+                BirthDate = user.BirthDate!.Value,
+                PhoneNumber = user.PhoneNumber!,
+                Employments = user.Employments
                     .Where(e => e.DateUntil == null)
                     .Select(e => new EmploymentVM
-                    {   
+                    {
                         EmploymentId = e.Id,
                         RestaurantId = e.RestaurantId,
                         IsBackdoorEmployee = e.IsBackdoorEmployee,
                         IsHallEmployee = e.IsHallEmployee,
-                        RestaurantName = e.Restaurant.Name,
+                        // Check for null Restaurant to avoid NullReferenceException
+                        RestaurantName = e.Restaurant != null ? e.Restaurant.Name : "Unknown",
                         DateFrom = e.DateFrom
                     })
-                    .ToList()
-            })
-            .ToListAsync();
-    }
-
-    /// <summary>
-    /// Gets the employee with the given id, provided he works for the current user
-    /// </summary>
-    /// <param name="empId"></param>
-    /// <param name="user"></param>
-    /// <returns></returns>
-    public async Task<Result<User>> GetEmployeeAsync(string empId, ClaimsPrincipal user)
-    {
-        var owner = (await userManager.GetUserAsync(user))!;
-        var emp = await userManager.FindByIdAsync(empId);
-
-        if (emp is null)
-        {
-            return new ValidationFailure
-            {
-                PropertyName = null,
-                ErrorMessage = $"Emp: {empId} not found",
-                ErrorCode = ErrorCodes.NotFound
+                    .ToList(),
+                Photo = uploadService.GetPathForFileName(user.PhotoFileName),
+                // Call GetFriendStatusAsync outside of the LINQ query
+                FriendStatus = await GetFriendStatusAsync(userId, user.Id)
```

Comment:
Myślę, że lepiej wstawić to wyżej w samej LINQ query (wzięto z `UserService.FindUsersAsync`)
```c#
                FriendStatus =
                    (from fr in dbContext.FriendRequests
                     let isOutgoing = fr.SenderId == currentUserId && fr.ReceiverId == u.Id
                     let isIncoming = fr.SenderId == u.Id && fr.ReceiverId == currentUserId
                     let isAccepted = fr.DateAccepted != null
                     where isOutgoing || isIncoming
                     select isAccepted
                        ? FriendStatus.Friend
                        : isIncoming
                        ? FriendStatus.IncomingRequest
                        : isOutgoing
                        ? FriendStatus.OutgoingRequest
                        : FriendStatus.Stranger)
```

To jest okropne powtórzenie kodu, ale w tym tygodniu planuję generalnie [rozwiązać](https://docs.automapper.org/en/stable/Getting-started.html) problem mapowania do DTO. Mimo to że powtarzamy się, zmniejsza ten kod liczbę zapytań do bazy danych (na razie jest N + 1) i ilość pobieranych danych